jbmc/src/java_bytecode/java_class_loader.cpp

```diff
@@ -71,21 +71,28 @@ java_class_loadert::parse_tree_with_overlayst &java_class_loadert::operator()(
 
 optionalt<java_bytecode_parse_treet> java_class_loadert::get_class_from_jar(
   const irep_idt &class_name,
-  const std::string &jar_file,
-  const jar_indext &jar_index)
+  const std::string &jar_file)
 {
-  auto jar_index_it = jar_index.find(class_name);
-  if(jar_index_it == jar_index.end())
-    return {};
+  const auto filename = class_name_to_file(class_name);
 
-  debug()
-    << "Getting class `" << class_name << "' from JAR " << jar_file << eom;
+  optionalt<std::string> data;
 
-  auto data = jar_pool(jar_file).get_entry(jar_index_it->second);
+  try
+  {
+    data = jar_pool(jar_file).get_entry(filename);
+  }
+  catch(const std::runtime_error &)
+  {
+    error() << "failed to open JAR file `" << jar_file << "'" << eom;
```

Comment:
Report exn.what()?

Reply:
We haven't done that yet anywhere -- before doing that, I'd like to first check whether these are useful to the user on our 3 main platforms.